components/TaskDetails.js

```diff
@@ -78,11 +79,18 @@ export default function TaskDetails({
   const isMissed = dueDate && checkForMissedDate(dueDate);
 
   async function handleDeleteTask(id) {
-    const response = await fetch(`/api/tasks/${id}`, {
-      method: "DELETE",
-    });
+    const response = await toast.promise(
+      fetch(`/api/tasks/${id}`, {
+        method: "DELETE",
+      }),
+      {
+        pending: "Task deletion is pending",
+        success: "Task deleted successfully",
+        error: "Task not deleted",
+      }
+    );
     if (response.ok) {
-      router.push("/");
+      router.push(`${detailsBackLinkRef}`);
```

Comment:
if `detailsBackLinkRef` is a string then i think we can just use it like so. 

```suggestion
      router.push(detailsBackLinkRef);
```

Reply:
Thank you for the review, suggestion implemented.